File: src/Tiantong.Wms/Tiantong.Wms.Client/src/Entities/OrderItem.ts
import OrderItemProject from './OrderItemProject'
import Invoice from './Invoice'
import { DateTime } from '@/utils/common'

export default class PurchaseOrderItem {
  id: number = 0

  order_id: number = 0

  good_id: number = 0

  item_id: number = 0

  index: number = 0

  price: number = 0

  quantity: number = 0

  comment: string = ''

  delivery_cycle: string = '30天'

  arrived_at: string = DateTime.now

  invoice: Invoice

  projects: Array<OrderItemProject>

  constructor (projects: Array<OrderItemProject> = []) {
    this.projects = projects
    this.invoice = new Invoice
  }
}

File: src/Tiantong.Wms/Tiantong.Wms.Api/Entities/OrderItem.cs
using System.Collections.Generic;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Tiantong.Wms.Api
{
  [Table("order_items")]
  public class OrderItem : Entity
  {
    public virtual int order_id { get; set; }

    public virtual int good_id { get; set; }

    public virtual int item_id { get; set; }

    public virtual int invoice_id { get; set; }

    public virtual int index { get; set; }

    public virtual double price { get; set; }

    public virtual int quantity { get; set; }

    public virtual string comment { get; set; }

    public virtual string delivery_cycle { get; set; }

    public virtual DateTime arrived_at { get; set; } = DateTime.Now;

    [ForeignKey("invoice_id")]
    public virtual Invoice invoice { get; set; }

    [ForeignKey("order_item_id")]
    public virtual List<OrderItemProject> projects { get; set; }
  }
}
